### PR TITLE
Tools: Use project language when scaffolding

### PR DESCRIPTION
**Pull request by bricelam**
Groundwork for #6401 & #8012

### CODE REVIEW

**Comment by bricelam** [comment]
@tonysneed and @ErikEJ beware. Let me know if this wreaks havoc on your extensions and, if so, how it can be improved.

**Comment by bricelam** [comment]
I did some manual bashing on this. F# projects in VS have some issues, but otherwise it looks like a good start.

Language     | PMC with .NET Framework | PMC with .NET Core | CLI
------------ | ----------------------- | ------------------ | ---
Visual Basic |                       :heavy_check_mark: |                  :heavy_check_mark: |   :heavy_check_mark:
F#           |                       :x: |                  :x: |   :heavy_check_mark:

File: src/EFCore.Tools/tools/EntityFrameworkCore.psm1 [hunk 1]
```diff
@@ -903,9 +905,11 @@ function IsDocker($project)
     return $project.Kind -eq '{E53339B2-1760-4266-BCC7-CA923CBCF16C}'
 }
 
-function IsCsproj2($project)
+function IsCpsProject($project)
 {
-    return $project.Kind -eq '{9A19103F-16F7-4668-BE54-9A1E7A4F7556}'
+    $hierarchy = GetVsHierarchy $project
+
+    return [Microsoft.VisualStudio.Shell.PackageUtilities]::IsCapabilityMatch($hierarchy, 'CPS')
```
Comment (bricelam) [inline src/EFCore.Tools/tools/EntityFrameworkCore.psm1]: @davkean Any idea why this check fails for F# projects?

Comment (bricelam) [reply src/EFCore.Tools/tools/EntityFrameworkCore.psm1]: Note, we do this before trying to [get the unconfigured project](https://github.com/Microsoft/VSProjectSystem/blob/master/doc/automation/finding_CPS_in_a_VS_project.md).

Comment (davkean) [reply src/EFCore.Tools/tools/EntityFrameworkCore.psm1]: F# isn't always CPS-based.

Comment (bricelam) [reply src/EFCore.Tools/tools/EntityFrameworkCore.psm1]: Turned out to be a limitation of PowerShell and NuGet; it was picking the (string, string) overload.

**Comment by bricelam** [comment]
Everything is working as expected now. EF Core ❤️s F# and Visual Basic!

**Comment by tonysneed** [comment]
I’ll give this a try with my stuff.

File: src/EFCore.Design/Migrations/Internal/MigrationsCodeGeneratorSelector.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        public MigrationsCodeGeneratorSelector([NotNull] IEnumerable<IMigrationsCodeGenerator> services)
+            : base(services)
+        {
+        }
+
+        /// <summary>
+        ///     This API supports the Entity Framework Core infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        public virtual IMigrationsCodeGenerator Override { get; [param: CanBeNull] set; }
```
Comment (smitpatel) [inline src/EFCore.Design/Migrations/Internal/MigrationsCodeGeneratorSelector.cs]: How would this Override used/set?

Comment (bricelam) [reply src/EFCore.Design/Migrations/Internal/MigrationsCodeGeneratorSelector.cs]: Only used to preserve the semantics of calling `MigrationsScaffolderDependencies.With(IMigrationsCodeGenerator)` can go away in 3.0.

**Comment by bricelam** [comment]
AFAIK, there aren't any braking changes to the public APIs.

**Pull request merged**